clients/src/test/java/org/apache/kafka/clients/consumer/internals/FetcherTest.java

```diff
@@ -316,6 +317,33 @@ public void testFetchNonContinuousRecords() {
         assertEquals(30L, consumerRecords.get(2).offset());
     }
 
+    /**
+     * Test the case where the client makes a FetchRequest, but the server replies with only a partial request.
+     * Prior to KIP-74, this would always happen when a single message was larger than the per-partition limit.
+     * The implementation of KIP-74 changed the behavior so that at least one message was always returned.
+     */
+    @Test
+    public void testFetchRequestWhenRecordTooBig() {
+        subscriptions.assignFromUser(singleton(tp));
+        subscriptions.seek(tp, 0);
+        assertEquals(1, fetcher.sendFetches());
+        assertFalse(fetcher.hasCompletedFetches());
+        MemoryRecords partialRecord = MemoryRecords.readableRecords(
+                ByteBuffer.wrap(new byte[]{0, 0, 0, 0, 0, 0, 0, 0}));
+        client.prepareResponse(fetchResponse(partialRecord, Errors.NONE.code(), 100L, 0));
+        consumerClient.poll(0);
+        assertTrue(fetcher.hasCompletedFetches());
+        try {
+            List<ConsumerRecord<byte[], byte[]>> consumerRecords = fetcher.fetchedRecords().get(tp);
+            assertEquals(0, consumerRecords.size());
+            fail("UnknownServerException should have been raised");
+        } catch (UnknownServerException e) {
```

Comment:
It seems more useful to test the `RecordTooLargeException` case here since it is the one we actually expect can happen in practice.